pyresample/geometry.py

```diff
@@ -185,29 +185,33 @@ def get_lonlat(self, row, col):
             raise ValueError('lon/lat values are not defined')
         return self.lons[row, col], self.lats[row, col]
 
-    def get_lonlats(self, data_slice=None, **kwargs):
+    def get_lonlats(self, data_slice=None, chunks=None, **kwargs):
         """Base method for lon lat retrieval with slicing"""
 
-        if self.lons is None or self.lats is None:
+        lons = self.lons
+        lats = self.lats
+        if lons is None or lats is None:
             raise ValueError('lon/lat values are not defined')
-        elif data_slice is None:
-            return self.lons, self.lats
-        else:
-            return self.lons[data_slice], self.lats[data_slice]
+        elif chunks is not None:
+            import dask.array as da
+            if isinstance(lons.data, da.Array):
+                # xarray DataArray
+                lons = lons.data
```

Comment:
I mean to the chunk size provided to the function could be used to rechunk the data. Don't know if that is a good idea though

Reply:
I think I'm starting to agree with you. If the lon/lats are already dask arrays then we should rechunk to what `chunks` is.

For DataArrays, yes it can accept them but should `get_lonlats` return the data underneath? Or should it return the DataArray? If it is a DataArray with a numpy array underneath and chunks is provided, do we return a DataArray with a dask array?